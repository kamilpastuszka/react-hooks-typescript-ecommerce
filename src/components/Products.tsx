import React, { useContext } from "react";
import Product from "./Product";
import { Context } from "../context/context";

export default function Products({ data }) {
  const { dispatch } = useContext(Context);

  const addProduct = (item: { id: number; name: string; price: number }) => {
    dispatch({ type: "ADD_PRODUCT", payload: item });
  };

  return (
    <div className="row d-flex justify-content-around">
      {data.map((item: { id: any; name: any; price: any }) => {
        return (
          <Product
            key={item.id}
            name={item.name}
            price={item.price}
            addProduct={() => addProduct(item)}
          />
        );
      })}
    </div>
  );
}
