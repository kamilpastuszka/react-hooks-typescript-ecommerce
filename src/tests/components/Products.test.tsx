import React from "react";
import { shallow } from "enzyme";
import Products from "../../components/Products";
import toJson from "enzyme-to-json";
import productData from "../fixtures/data";

test("should render Basket correctly", () => {
  const wrapper = shallow(<Products data={productData} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});
