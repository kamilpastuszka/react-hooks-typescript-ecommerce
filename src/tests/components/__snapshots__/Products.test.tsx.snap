// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Basket correctly 1`] = `
<div
  className="row d-flex justify-content-around"
>
  <Product
    addProduct={[Function]}
    key="1"
    name="Item 1"
    price={100.99}
  />
  <Product
    addProduct={[Function]}
    key="2"
    name="Item 2"
    price={200.99}
  />
  <Product
    addProduct={[Function]}
    key="3"
    name="Item 3"
    price={300.99}
  />
  <Product
    addProduct={[Function]}
    key="4"
    name="Item 4"
    price={499.99}
  />
  <Product
    addProduct={[Function]}
    key="5"
    name="Item 5 "
    price={500.99}
  />
  <Product
    addProduct={[Function]}
    key="6"
    name="Item 6"
    price={600.99}
  />
</div>
`;
